.github/scripts/utils.py/update-crowdin-locales.py

```diff
@@ -0,0 +1,88 @@
+import os
+import logging
+import json
+
+from crowdin_api import CrowdinClient
+from fnmatch import fnmatch
+
+CROWDIN_PROJECT_ID = 51028
+CROWDIN_API_TOKEN = os.environ["CROWDIN_PAT"]
+DEFAULT_LOCALE = "en-US"
+
+crowdin_api = CrowdinClient(token=CROWDIN_API_TOKEN, project_id=CROWDIN_PROJECT_ID, page_size=500)
+
+logging.basicConfig()
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def get_dir_id(name: str) -> int:
+    crowdin_api.source_files.list_directories(filter=name)["data"][0]["data"]["id"]
+
+
+def add_to_crowdin_storage(file_path: str) -> int:
+    return crowdin_api.storages.add_storage(open(file_path))["data"]["id"]
```

Comment:
See above re. avoidance of `ValueError`s.

Reply:
When I try...except this theses object accesses, what should I do in the except block? I caught the error, but should I raise an exception again? Just log it?

My only idea would be to try...except the upload() function in the main function (https://github.com/Vainock/obs-studio/blob/ba977d75bd6f8e1546f0d2824d6714110073af31/.github/scripts/utils.py/update-crowdin-locales.py#L101) for all `KeyError`s and `ValueError`s. This would cover all instances where these exceptions can occur in that file. My problem is that I can only return the error code from the main function. Do you have an idea for how to solve that?

Reply:
You don't have to catch exceptions right where they happen but where you can handle them, so it'd be fine to not handle them on lower levels and catch them on the top-most level and log the error there and return from the main function with an error code.

The exception to that would be if you want to be able to more specifically identify the error to have more specific error messages (so you want to differentiate _which_ interaction resulted in a `ValueError`). This would also enable you to `raise` errors with error messages in the inner functions instead of having to log error messages, because you're catching them in `main` and print the error there.